tests/Tokenizer/Analyzer/FunctionsAnalyzerTest.php

```diff
@@ -686,17 +686,25 @@ public static function provideFunctionReturnTypeInfoPre80Cases(): iterable
 
     /**
      * @dataProvider provideIsTheSameClassCallCases
+     *
+     * @param list<int> $sameClassCallIndices
      */
-    public function testIsTheSameClassCall(bool $isTheSameClassCall, string $code, int $index): void
+    public function testIsTheSameClassCall(string $code, array $sameClassCallIndices): void
     {
         $tokens = Tokens::fromCode($code);
         $analyzer = new FunctionsAnalyzer();
 
-        self::assertSame($isTheSameClassCall, $analyzer->isTheSameClassCall($tokens, $index));
+        for ($index = $tokens->count(); $index >= 0; --$index) {
```

Comment:
```suggestion
        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
```
if we have size==2, then `array[2]` doesn't exist

Reply:
Exactly: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.49.0/src/Tokenizer/Analyzer/FunctionsAnalyzer.php#L186-L188

Previous tests had checked for `-1`: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.49.0/tests/Tokenizer/Analyzer/FunctionsAnalyzerTest.php#L711-L715

Maybe we want to throw an error for a non-existing index? Definitely, we want to check what will happen if the non-existing index is given.

Reply:
to me code looks very counterintuitive to iterate over all existing indexes plus one non-existing , out of the blue it looks almost like a typo error to be fixed (even by sca detecting calling `$array[non-exist-index]`, if sca advanced enough).

i would encourage to have dedicated check for non-existing index.

Reply:
> Maybe we want to throw an error for a non-existing index?

out of the blue sounds like possible variant, probably how I would drive it if creating this function from scratch [~indexoutofboundsexception].
we could check how code would react, i hope we do not rely on this existing behaviour to return false instead of error

Reply:
Done. based on https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.49.0/src/Tokenizer/Analyzer/NamespacesAnalyzer.php#L77